### ISSUE TITLE
Experiment: learned unknown bias on union_name and f_num heads

### ISSUE THREAD

**Issue by fgregg** [labordata/labor-union-parse#9]
## Idea

Add a learned scalar bias (no embedding) to the union_name and f_num classification heads that represents "unknown." Instead of a learned embedding vector competing in dot-product space (which can't represent "not like anything"), use a constant logit `c` that competes with the real class logits in the softmax:

```python
logits_real = W @ h + b          # input-dependent, as usual
logit_unknown = c                # learned scalar, input-independent
all_logits = cat(logits_real, logit_unknown)
probs = softmax(all_logits)
```

When no real class gets a strong logit, `c` wins and the model predicts "unknown." This probability mass on "unknown" naturally lowers P(real_class) for every class, which lowers gazetteer record scores during factored scoring, making null_bias more likely to win.

## Motivation

The structured classifier currently only sees positive examples — every training input has a known match. It never learns what "no match" looks like, so at inference on unmatched inputs it confidently picks the closest-looking class, producing false positive evidence that the scoring layer's null_bias has to overcome alone.

The ~23K no-f_num training examples could provide training signal for this. For f_num, all 23K would be labeled "unknown." For union_name, the 79 truly-unknown-union examples would be labeled "unknown" (the other 23K have known union names).

## Comparison with is_union detector

This could potentially replace or improve on the separate is_union detector, since the structured classifier learns much richer representations of union text structure. The unknown bias would leverage that learned structure to detect "not a union" or "not a known f_num" as a byproduct of classification, rather than needing a separate model.

## Implementation

- Add a `nn.Parameter(torch.zeros(1))` for each head that gets the unknown bias
- Concatenate it to the logits before softmax
- Label no-f_num examples as "unknown" for f_num head (23K examples)
- Label unknown-union examples as "unknown" for union_name head (79 examples)
- Train with standard cross-entropy (unknown is just another class index)
- At scoring time, P(unknown) reduces all real class probs — no special handling needed

**Comment by fgregg** [comment]
## Results: Slight Regression

Added learned scalar `unknown_bias` to `union_name` and `f_num` classification heads. The bias competes with real class logits in softmax — when no real class is strong, the unknown class should win.

### Training
- **f_num unknown**: 4,667 examples (reasons: not in gazetteer, unknown union, not a union). "Potentially resolvable" kept as ignored (-100).
- **union_name unknown**: ~87 examples (reasons: unknown union, not a union).
- Learned bias values: f_num=0.768, union_name=0.461

### End-to-end evaluation
| Metric | Baseline | This experiment | Delta |
|--------|----------|----------------|-------|
| E2E accuracy | 97.92% (160 errors) | 97.79% (170 errors) | -0.13% |
| Wrong f_num matches | ~135 | 168 | +33 |
| Precision@0.30 | 98.91% (77 errors) | 98.86% (80 errors) | -0.05% |

Per-field regressions: desig_name 55→68, f_num 115→135, desig_num 80→106.

### Conclusion
Negative result. Closing as won't-fix.

Branch: `bias_experiment` (commit d8f6362)
